alf/examples/networks/impala_cnn_encoder.py

```diff
@@ -0,0 +1,186 @@
+            after the ``Conv2D`` and the downsampling max pooling filter.
+        kernel_initializer (Callable[[Tensor], None]): initializer for
+            the Conv2D and FC layers in the module.
+
+    Returns:
+        
+        A module that perofrms the described operations.
+
+    """
+    input_channels = input_tensor_spec.shape[0]
+    # The output tensor spec is indentical to the input tesnor spec except for
+    # that the channel dimension is replaced by the output number of channels.
+    output_tensor_spec = TensorSpec(
+        (output_channels, *input_tensor_spec.shape[1:]),
+        input_tensor_spec.dtype)
+    return alf.nn.Sequential(
+        # NOTE: this Conv2D layer does not have activation
+        alf.layers.Conv2D(
+            input_channels,
+            output_channels,
+            kernel_size=3,
+            padding=1,
+            activation=identity,
+            kernel_initializer=kernel_initializer),
+        # Downsample via a max pooling filter
+        torch.nn.MaxPool2d(kernel_size=3, stride=2, padding=1),
+        *[
+            _create_residual_cnn_block(
+                output_tensor_spec, kernel_initializer=kernel_initializer)
+            for _ in range(num_residual_blocks)
+        ],
+        input_tensor_spec=input_tensor_spec)
+
+
+def create(input_tensor_spec,
+           cnn_channel_list: List[int] = (16, 32, 32),
+           num_blocks_per_stack: int = 2,
+           output_size: int = 256,
+           kernel_initializer=torch.nn.init.xavier_uniform_):
+    """Create the Impala CNN Encoder
+
+    Here the so called Impala CNN Encoder is essentially a series of CNN
+    downsampling stacks with an FC layer at the end to output an 1D tensor (if
+    not counting the batch dimension) of the specified output size.
+
+    Note that the default values for the parameters are the ones that are used
+    for the procgen environments. Credits to the OpenAI's official PPG
+    implementation.
+
+    If your use case is different, it is very likely that you will need to find
+    your own parameters.
+
+    Args:
+
+        input_tensor_spec (nested TensorSpec): The spec of the input tensor,
+            denotated as ``x`` in the above formula. The shape of the input
+            tensor omitting batch size should be (C, W, H), where C is the input
+            number of channels.
+        cnn_channel_list (List[int]): a list that specifies the number of output
+            channels of each CNN downsampling stack in the application order.
+        num_blocks_per_stack (int): specifies how many residual blocks to apply
+            in each of the CNN downsampling stack. This parameter is uniform to
+            all the CNN downsampling stacks.
+        output_size (int): the size of the output vector for each of the input
+            image in the mini batch.
+        kernel_initializer (Callable[[Tensor], None]): initializer for
```

Comment:
It seems that this format on type (`Callable[[Tensor], None]`) is a bit different from existing ones. Maybe can consider using something like ``Callable[Tensor] | None``?

Also for Line 42.

Reply:
Actually `Callable[[Tensor], None]` means it is a function like `f(x: Tensor) -> None`. This is python's type hints standard, while `Callable | None` means `Optional[Callable]`, meaning that the argument could be `None`.

Here the type hint is used to indicate the former.

Reply:
> Actually `Callable[[Tensor], None]` means it is a function like `f(x: Tensor) -> None`. This is python's type hints standard, while `Callable | None` means `Optional[Callable]`, meaning that the argument could be `None`.
> 
> Here the type hint is used to indicate the former.

Understand. What I meant was indeed the latter case. Actually, if we pass a `None` value to `kernel_initializer`, a variance scaling based initializer will be used.

Reply:
Ah I see! Sorry that I misunderstood your comments. I agree with you that the consistent interface would be better. In this case at least the public interface `create` is better to be consistent. Will update that in another PR. Appreciate the explanation!